### PR TITLE
LCAM 1105 Integration Tests for Means Assessment Flow

### PR DESCRIPTION
**Pull request by nigelpain**
## What

[Link to story](https://dsdmoj.atlassian.net/jira/software/projects/LCAM/boards/881?selectedIssue=LCAM-1105)

Changed CrownCourtOutcomeType enum to be public so it can be accessed from any project that uses this module (in particular orchestration for a fix highlighted by the integration tests) and also added unit tests for the new enum.

### CODE REVIEW

**Review by skasthuri36326 — DISMISSED** [review]

File: crime-commons-classes/src/main/java/uk/gov/justice/laa/crime/enums/CrownCourtOutcomeType.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    private final String type;
+
+    public static CrownCourtOutcomeType getFrom(String type) {
+        if (StringUtils.isBlank(type)) return null;
+
+        return Stream.of(CrownCourtOutcomeType.values())
+                .filter(ccoType -> ccoType.type.equals(type))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(String.format("CrownCourtOutcomeType with value: %s does not exist.", type)));
```
Comment (stone-dj) [inline crime-commons-classes/src/main/java/uk/gov/justice/laa/crime/enums/CrownCourtOutcomeType.java:22]: Personally I have a preference for enclosing 'untrusted' or otherwise invalid input in messages as I feel it is easier to spot some types of issues. It's a pretty edge case, so totally optional, but something to consider!

i.e. consider the following
```
    @Test
    void givenInvalidResultString_whenGetFromIsInvoked_thenExceptionIsThrown() {
       CrownCourtOutcome.getFrom("MOCK_RESULT_STRING exists\n\n\n\n\n\n\n");
    }
```
Will throw an exception with the message:
```
java.lang.IllegalArgumentException: Crown Court Outcome with value: MOCK_RESULT_STRING exists






 does not exist.
```
Which if not properly escaped (and/oror enclosed) could be misleading at first glance, compared to
```
java.lang.IllegalArgumentException: Crown Court Outcome with value: [MOCK_RESULT_STRING exists






] does not exist.
```

Comment (nigelpain) [reply crime-commons-classes/src/main/java/uk/gov/justice/laa/crime/enums/CrownCourtOutcomeType.java]: Ok a useful thing to remember thanks! Enclosed the untrusted input in the exception message above 👍

**Review by venkat980 — APPROVED** [review]

**Review by stone-dj — APPROVED** [review]
